### PR TITLE
Login/register forms, Creating users, started auth

### PR DESCRIPTION
**Pull request by Bat0oo**
I did login/register forms, started users controller and authentication, routes, navbar
Didn't have time to finish it all, but wanted to commit it so it can be reviewed 

Edit:
new model, new migration, new table

### CODE REVIEW

File: Konteh/Konteh.BackOfficeApi/Features/Users/CreateUser.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public class Command : IRequest
+    {
+        public string Name { get; set; } = string.Empty;
+        public string LastName { get; set; } = string.Empty;
+        public string Email { get; set; } = string.Empty;
+        public string Password { get; set; } = string.Empty;
+    }
+    public class RequestHandler : IRequestHandler<Command>
+    {
+        private readonly IRepository<User> _userRepository;
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Features/Users/CreateUser.cs:18]: Oh no. We don't need any of this. Only Novalite employees will be using this, and we have different ways of keeping track of them

File: Konteh/Konteh.BackOfficeApi/Program.cs [hunk 2]
```diff
@@ -11,6 +11,8 @@
 
 builder.Services.AddOpenApiDocument(options => options.SchemaSettings.SchemaNameGenerator = new CustomSwaggerSchemaNameGenerator());
 
+builder.Configuration.AddJsonFile("appsettings.json");
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Program.cs:14]: What is this for?

File: Konteh/Konteh.FrontOfficeApi/Konteh.FrontOfficeApi.csproj [hunk 3]
```diff
@@ -19,6 +19,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.61.0" />
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOfficeApi/Konteh.FrontOfficeApi.csproj:22]: Not sure why this package was added here.

File: Konteh/Konteh.FrontOfficeApi/appsettings.json [hunk 4]
```diff
@@ -8,5 +8,11 @@
   "ConnectionStrings": {
     "KontehDB": "Server=.;Database=KontehDB;Trusted_Connection=True;MultipleActiveResultSets=true;TrustServerCertificate=True;"
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOfficeApi/appsettings.json:11]: Frontoffice will not be using EntraID, remove this please

File: Konteh/Konteh.Infrastructure/Migrations/20240517115523_AddedUsers.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
```
Comment (laslorajda) [inline Konteh/Konteh.Infrastructure/Migrations/20240517115523_AddedUsers.cs:3]: remove this migration

Comment (laslorajda) [reply Konteh/Konteh.Infrastructure/Migrations/20240517115523_AddedUsers.cs:3]: google how to do that via the CLI

File: back-office-app/src/app/app-routing.module.ts [hunk 6]
```diff
@@ -1,7 +1,13 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
+import { LoginComponent } from './login/login.component';
+import { RegisterComponent } from './register/register.component';
+import { NavbarComponent } from './navbar/navbar.component';
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path: 'login', component: LoginComponent},
+  {path: 'register',component:RegisterComponent},
```
Comment (laslorajda) [inline back-office-app/src/app/app-routing.module.ts:9]: We won't have any registration

File: back-office-app/src/app/config.service.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConfigService {
```
Comment (laslorajda) [inline back-office-app/src/app/config.service.ts:6]: I thought we moved this to environment files?

File: back-office-app/src/custom-theme.scss [hunk 8]
```diff
@@ -0,0 +1,37 @@
+
+// Custom Theming for Angular Material
+// For more information: https://material.angular.io/guide/theming
+@use '@angular/material' as mat;
+// Plus imports for other components in your app.
+
+// Include the common styles for Angular Material. We include this here so that you only
```
Comment (laslorajda) [inline back-office-app/src/custom-theme.scss:7]: the what?

File: back-office-app/src/app/app.component.ts [hunk 9]
```diff
@@ -7,4 +10,10 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'back-office-app';
+
+  constructor(private authService: AuthService, private router: Router){}
+
+  navigateToForm2() {
```
Comment (laslorajda) [inline back-office-app/src/app/app.component.ts:16]: what does any of this do?

File: back-office-app/src/app/navbar/navbar.component.html [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<button mat-button class="mat-button" (click)="navigateTo('')">Home</button>
+<div class="button-container">
+    <button mat-button (click)="navigateTo('')">Home</button>
```
Comment (laslorajda) [inline back-office-app/src/app/navbar/navbar.component.html:3]: Just tested these buttons, they work perfectly fine.

File: back-office-app/src/app/login/login.component.html [hunk 11]
```diff
@@ -0,0 +1,24 @@
+<div class="login-container">
+  <h2>Login</h2>
+  <form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
```
Comment (laslorajda) [inline back-office-app/src/app/login/login.component.html:3]: noooo, just use the default Microsoft stuff, all this work was unnecessary. In the future if you are unsure about what to do, please ask.

File: Konteh/Konteh.BackOfficeApi/Controllers/QuestionsController.cs [hunk 12]
```diff
@@ -19,8 +21,19 @@ public QuestionsController(IMediator mediator)
     [HttpGet]
     public async Task<ActionResult<IEnumerable<GetAllQuestions.Response>>> GetAll()
     {
-        var result = await _mediator.Send(new GetAllQuestions.Query());
-        return Ok(result);
+        try
+        {
+
```
Comment (irenva101) [inline Konteh/Konteh.BackOfficeApi/Controllers/QuestionsController.cs:26]: delete spacing please

**Review by irenva101 — APPROVED** [review]

File: Konteh/Konteh.BackOfficeApi/Controllers/QuestionsController.cs [hunk 13]
```diff
@@ -19,8 +21,19 @@ public QuestionsController(IMediator mediator)
...
+            var result = await _mediator.Send(new GetAllQuestions.Query());
+            var userId = HttpContext.User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Controllers/QuestionsController.cs:28]: what is this for?

File: Konteh/Konteh.BackOfficeApi/Controllers/QuestionsController.cs [hunk 14]
```diff
@@ -19,8 +21,19 @@ public QuestionsController(IMediator mediator)
...
+            string token = Request.Headers.Authorization.ToString();
+
+            return Ok(result);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Erorr occurred while processing your request {ex}");
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Controllers/QuestionsController.cs:35]: I don't see what value this adds

File: Konteh/Konteh.BackOfficeApi/Konteh.BackOfficeApi.csproj [hunk 15]
```diff
@@ -20,6 +22,8 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.61.0" />
```
Comment (laslorajda) [inline Konteh/Konteh.BackOfficeApi/Konteh.BackOfficeApi.csproj:25]: you don't need this dependency

File: Konteh/Konteh.Domain/Konteh.Domain.csproj [hunk 16]
```diff
@@ -6,4 +6,9 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="8.0.5" />
```
Comment (laslorajda) [inline Konteh/Konteh.Domain/Konteh.Domain.csproj:10]: Why are these dependencies in the domain project?!

File: Konteh/Konteh.FrontOfficeApi/Program.cs [hunk 17]
```diff
@@ -12,18 +12,20 @@
...
 
+var app = builder.Build();
+
 app.UseOpenApi();
 app.UseSwaggerUi();
 
 app.UseHttpsRedirection();
+app.UseRouting();
 
+app.UseAuthentication();
```
Comment (laslorajda) [inline Konteh/Konteh.FrontOfficeApi/Program.cs:28]: you can probably remove these from FrontOffice

File: Konteh/Konteh.Infrastructure/Konteh.Infrastructure.csproj [hunk 18]
```diff
@@ -7,9 +7,11 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="8.0.5" />
```
Comment (laslorajda) [inline Konteh/Konteh.Infrastructure/Konteh.Infrastructure.csproj:10]: These packages do not belong here.

File: back-office-app/src/app/app.component.ts [hunk 19]
```diff
@@ -1,10 +1,33 @@
...
-export class AppComponent {
+export class AppComponent implements OnInit {
   title = 'back-office-app';
+  isIframe=false;
+  loginDisplay=false;
+
+  constructor(private authService:MsalService, private msalBroadcastService: MsalBroadcastService){}
+
+  ngOnInit(): void {
+    this.msalBroadcastService.msalSubject$
```
Comment (laslorajda) [inline back-office-app/src/app/app.component.ts:19]: please delete all of this, I'm pretty sure it's unused

File: back-office-app/src/app/auth.service.ts [hunk 20]
```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { PublicClientApplication, AuthenticationResult } from '@azure/msal-browser';
+import {environment} from '../environments/environment';
+import { MsalModule, MsalService } from '@azure/msal-angular';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
```
Comment (laslorajda) [inline back-office-app/src/app/auth.service.ts:10]: I'm not sure you need this service?

File: back-office-app/src/app/navbar/navbar.component.html [hunk 21]
```diff
@@ -0,0 +1,6 @@
+<mat-toolbar color="primary">
+    <button mat-button (click)="navigateTo('')">Home</button>
+    <button mat-button *ngIf="!loginDisplay" (click)="login()">Login</button>
+    <button mat-button *ngIf="loginDisplay" (click)="logout()">Logout</button>
+    <button mat-button (click)="test123()">Test123</button>
```
Comment (laslorajda) [inline back-office-app/src/app/navbar/navbar.component.html:5]: please delete this, it was used for testing

File: back-office-app/src/app/navbar/navbar.component.ts [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+          console.error('Redirecr error: ', error)
+        }
+      });
+    this.setLoginDisplay();
+  }
+  setLoginDisplay() {
+    this.loginDisplay = this.authService.instance.getAllAccounts().length > 0;
+  }
+
+  private idTokenKey='idToken';
```
Comment (laslorajda) [inline back-office-app/src/app/navbar/navbar.component.ts:42]: What does this do?

File: back-office-app/src/app/navbar/navbar.component.ts [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+  login() {
+    this.authService.loginPopup()
+      .subscribe((response: AuthenticationResult) => {
+        localStorage.setItem(this.idTokenKey, response.idToken);
+        this.authService.instance.setActiveAccount(response.account);
+        this.loginDisplay = true;
+        console.log(response.account);
```
Comment (laslorajda) [inline back-office-app/src/app/navbar/navbar.component.ts:49]: no console logs please

File: back-office-app/src/app/navbar/navbar.component.ts [hunk 24]
```diff
@@ -0,0 +1,83 @@
...
+  loginDisplay: boolean = false;
+  isIframe = false;
+  private readonly _destroying$ = new Subject<void>();
+
+  constructor(
+    private authService: MsalService,
+    private router: Router,
+    private client: QuestionsClient
+  ) {
+    this.authService.handleRedirectObservable()
```
Comment (laslorajda) [inline back-office-app/src/app/navbar/navbar.component.ts:23]: pretty sure you can get rid of all of this.

File: back-office-app/src/environments/environment.development.ts [hunk 25]
```diff
@@ -0,0 +1,9 @@
+export const environment = {
+    clientId: '4e1ff54b-bf34-4f45-83ce-e50fc32967cd',
+  authority: 'https://login.microsoftonline.com/common',
+  redirectUri: 'http://localhost:4200',
+  scopes: [
+    'api://dbf7f51e-d046-435b-88ee-c4f9ee872967/to-do-lists.read',
+    'api://dbf7f51e-d046-435b-88ee-c4f9ee872967/to-do-lists.write'
+  ]
```
Comment (laslorajda) [inline back-office-app/src/environments/environment.development.ts:8]: formatting

**Review by laslorajda — CHANGES_REQUESTED** [review]

**Review by laslorajda — APPROVED** [review]

**Pull request merged**